sympy/integrals/laplace.py

```diff
@@ -576,6 +577,15 @@ def _laplace_rule_heaviside(f, t, s):
             _debug('     rule: Heaviside factor, negative time shift (4.1.4)')
             r, pr, cr = _laplace_transform(ma1[g], t, s, simplify=False)
             return (r, pr, cr)
+        ma2 = ma1[y].match(a-t)
+        if ma2 and ma2[a].is_positive:
+            _debug('     rule: Heaviside window open')
+            r, pr, cr = _laplace_transform(
+                (1-Heaviside(t-ma2[a]))*ma1[g], t, s, simplify=False)
+            return (r, pr, cr)
+        if ma2 and ma2[a].is_negative:
+            _debug('     rule: Heaviside window closed')
+            return (0, 0, S.true)
```

Comment:
```suggestion
        if ma2 := ma1[y].match(a - t):
            if ma2[a].is_positive:
                _debug('     rule: Heaviside window open')
                r, pr, cr = _laplace_transform(
                    (1 - Heaviside(t - ma2[a]))*ma1[g], t, s, simplify=False)
                return (r, pr, cr)
            if ma2[a].is_negative:
                _debug('     rule: Heaviside window closed')
                return (0, 0, S.true)
```

Reply:
Nice, thanks! I will also alter my other change in this PR accordingly. Long live the Walrus operator ;)